tests/tt_metal/tt_metal/perf_microbenchmark/dispatch/test_pgm_dispatch.cpp

```diff
@@ -222,25 +222,24 @@ int main(int argc, char **argv) {
         initialize_program(program[1], fast_kernel_cycles_g);
 
         // Cache stuff
+        EnqueueProgram(cq, program[0], false);
+        EnqueueProgram(cq, program[1], false);
+        uint32_t tid = BeginTraceCapture(device, cq.id());
         for (int i = 0; i < warmup_iterations_g; i++) {
             EnqueueProgram(cq, program[0], false);
             if (nfast_kernels_g > 0) {
                 EnqueueProgram(cq, program[1], false);
             }
         }
```

Comment:
When we capture the warmup in the trace, we run each program for 100 iterations when we enqueue the trace.
The original behaviour of the test is to profile host dispatch for `iterations_g`, which is 10000.

To ensure a 1:1 comparison between traced and untraced runs, we should capture the loop over `iterations_g` in the trace.

Reply:
Right, makes sense.